File: mezzang/book/08.js
function solution(s) {
    const stack = [];
    for(const c of s){
        if(c === '('){
            stack.push(c);
        } else if (c === ')'){
            if (stack.length === 0){
                return false;
            }else {
                stack.pop();
            }
        }
    }
    return stack.length === 0; //boolean 값 리턴
}

console.log(solution(' (())()'));

File: seung365/programmers/08.py

def check(exam):
  stack = []

  for char in exam:
    if char == '(':
      stack.append(char)
    else:
      if stack[-1] == '(':
        stack.pop()
      else:
        return False
  
  if len(stack) > 0:
    return False
  else:
    return True  

print(check('(())()'))
print(check('((())())'))
print(check('((())()'))

